### ISSUE TITLE
re-enable `sort_directives` lint pending new linter release

### ISSUE THREAD

**Issue by pq** [flutter/flutter#84577]
The `sort_directives` lint has recently been fixed to be compatible w/ the Dart Analysis Server's (DAS) "sort Dart members" action (https://github.com/dart-lang/linter/pull/2688).  This is a big win and will allow DAS to be used for sorting, obviating the need for hand-rolled solutions (notably used internally).  Unfortunately, there are a few places where the new order is incompatible w/ the old and so landing the change requires coordination.

Here are the current breakages:

https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8844442836428066848/+/u/analyze_flutter_flutter/stdout

```
   info • Sort directive sections alphabetically • dev/benchmarks/microbenchmarks/lib/gestures/gesture_detector_bench.dart:8:1 • directives_ordering
   info • Sort directive sections alphabetically • dev/conductor/test/candidates_test.dart:12:1 • directives_ordering
   info • Sort directive sections alphabetically • dev/conductor/test/clean_test.dart:15:1 • directives_ordering
   info • Sort directive sections alphabetically • dev/conductor/test/codesign_test.dart:14:1 • directives_ordering
   info • Sort directive sections alphabetically • dev/conductor/test/repository_test.dart:12:1 • directives_ordering
   info • Sort directive sections alphabetically • dev/conductor/test/roll_dev_test.dart:14:1 • directives_ordering
   info • Sort directive sections alphabetically • dev/conductor/test/start_test.dart:20:1 • directives_ordering
```

(Engine try is still rolling.)

I can see a few options:

1. add an `analysis_options.yaml` file to `dev` that disables the lint there
2. add ignores to the offending lines (ugh)
3. disable the lint globally (in root options)

All with an eye towards tidying up once the new lint is integrated.

My vote would be for option 1 but I'd love to get input from other folks.

Thanks in advance!

/cc @goderbauer @jonahwilliams @Hixie @davidmorgan @devoncarew

**Comment by pq** [comment]
PR that takes approach 1: https://github.com/flutter/flutter/pull/84579/

**Comment by pq** [comment]
After a revert of `1.6.0`, the relevant change landed in `1.6.1` w/ https://github.com/dart-lang/sdk/commit/b5e76568336729efe431422487a22eb3f78073d3.  When that makes it's way into Flutter, I'll go ahead and re-enable.
